buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/PerformanceTest.groovy

```diff
@@ -14,109 +14,144 @@
  * limitations under the License.
  */
 
-package org.gradle.testing;
+package org.gradle.testing
 
-import org.gradle.api.internal.tasks.options.Option;
-import org.gradle.api.tasks.Input;
-import org.gradle.api.tasks.Optional;
-import org.gradle.api.tasks.OutputDirectory;
-import org.gradle.gradlebuild.test.integrationtests.DistributionTest;
+import org.gradle.api.internal.tasks.options.Option
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.gradlebuild.test.integrationtests.DistributionTest
 
-import javax.annotation.Nullable;
-import java.io.File;
+import javax.annotation.Nullable
+import org.gradle.api.Action
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.process.JavaExecSpec
+import groovy.transform.CompileStatic
 
 /**
  * A test that checks execution time and memory consumption.
  */
+@CacheableTask
+@CompileStatic
 public class PerformanceTest extends DistributionTest {
+    @Nullable @Optional @Input
+    String baselines
+
+    @Nullable @Optional @Input
+    String warmups
 
-    private String baselines;
-    private String warmups;
-    private String runs;
-    private String checks;
-    private String channel;
-    private File debugArtifactsDirectory = new File(getProject().getBuildDir(), getName());
+    @Nullable @Optional @Input
+    String runs
+
+    @Nullable @Optional @Input
+    String checks
+
+    @Nullable @Optional @Input
+    String channel
+
+    @Input
+    String resultStoreClass = "org.gradle.performance.results.AllResultsStore"
+
+    @Input
+    boolean generatePerformanceReport = true
```

Comment:
Well, it is, but I think here I think it should be `true` by default, because basically we have 3 scenarios: 

- distributed performance test which requires report generation.
- individual performance test which doesn't require report generation.
- adhoc performance test which requires report generation.

So "in the most cases" we need report generated - that's what we're doing now: enable `peformanceReport` task by default and only disable it for individual performance test scenario.

Reply:
Where are you disabling the report for the individual performance test? Currently we don't have reports for the ad hoc performance runs (at least on CI), that's why I thought we would disable it by default.

Reply:
Okay, I confused flame graph and performance report. Then yes, the default should be `false`.

Reply:
Based on these analysis, I removed `generatePerformanceReport` property and let `PerformanceTest` return `false` and `DistributedPerformanceTest` return `true`.